### PR TITLE
[Merged by Bors] - feat(AlgebraicGeometry): smooth morphisms are locally of finite presentation

### PR DESCRIPTION
**Pull request by chrisflav**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/AlgebraicGeometry/Morphisms/Smooth.lean [hunk 1]
```diff
@@ -165,4 +166,17 @@ instance : MorphismProperty.IsMultiplicative (@IsSmoothOfRelativeDimension 0) wh
...
+instance (priority := 100) [hf : IsSmooth f] : LocallyOfFinitePresentation f := by
+  rw [HasRingHomProperty.eq_affineLocally @LocallyOfFinitePresentation]
+  rw [HasRingHomProperty.eq_affineLocally @IsSmooth] at hf
+  refine affineLocally_le (fun hf ↦ ?_) f hf
+  apply RingHom.locally_of_locally (Q := RingHom.FinitePresentation) at hf
+  · rwa [RingHom.locally_iff_of_localizationSpanTarget finitePresentation_respectsIso
+      finitePresentation_ofLocalizationSpanTarget] at hf
+  · introv hf
+    algebraize [f]
+    haveI : Algebra.IsStandardSmooth R S := hf
```
Comment (chrisflav) [inline Mathlib/AlgebraicGeometry/Morphisms/Smooth.lean:179]: @callesonne do you understand why `algebraize` does not generate this instance?

File: Mathlib/AlgebraicGeometry/Morphisms/Smooth.lean [hunk 2]
```diff
@@ -165,4 +166,17 @@ instance : MorphismProperty.IsMultiplicative (@IsSmoothOfRelativeDimension 0) wh
   id_mem _ := inferInstance
   comp_mem _ _ _ _ := inferInstance
 
+/-- Smooth morphisms are of finite presentation. -/
```
Comment (joelriou) [inline Mathlib/AlgebraicGeometry/Morphisms/Smooth.lean:169]:
```suggestion
/-- Smooth morphisms are locally of finite presentation. -/
```

File: Mathlib/AlgebraicGeometry/Morphisms/Smooth.lean (hunk 1 above)
Comment (jcommelin) [reply Mathlib/AlgebraicGeometry/Morphisms/Smooth.lean:179]: LGTM, modulo this issue. So maybe
```suggestion
    algebraize [f]
    -- TODO: why is `algebraize` not generating the following instance?
    haveI : Algebra.IsStandardSmooth R S := hf
```

**Review by jcommelin — COMMENTED** [review]
maintainer delegate

**Comment by joelriou** [comment]
Thanks!

bors merge

**Pull request closed without merging**